### ISSUE TITLE
Inline arrow function inside constructor misinterpreted as constructor args

### ISSUE THREAD

**Issue by samdenty** [Microsoft/TypeScript#29286]
**TypeScript Version:**  3.3.0-dev.201xxxxx

**Search Terms:**

'this' cannot be referenced in constructor arguments.

**Code**

```ts
class Test {
  constructor() {
    const func = (a = this) => 1
  }
}
```

**Expected behavior:**

Typescript shouldn't produce any errors, as `this` is not referenced inside the constructors parameters.

**Actual behavior:**

Typescript misinterprets the default parameter of the inline arrow function as a default parameter of the constructor (which is not allowed, and should produce an error).

**Playground Link:** [here](https://www.typescriptlang.org/play/index.html#src=class%20Test%20%7B%0D%0A%20%20constructor()%20%7B%0D%0A%20%20%20%20%3B%20(a%20%3D%20this)%20%3D>%201%0D%0A%20%20%7D%0D%0A%7D)

**Related Issues:** None

**Comment by ajafff** [comment]
Fix is up at #29298
